httputil/client.go

```diff
@@ -33,6 +41,55 @@ type ClientOptions struct {
 	Proxy      func(*http.Request) (*url.URL, error)
 }
 
+func addLocalSslCertificates(conf *tls.Config) (allCAs *x509.CertPool, err error) {
+	if conf != nil && conf.RootCAs != nil {
+		allCAs = conf.RootCAs
+	} else {
+		allCAs, err = x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("cannot read system certificates: %v", err)
+		}
+	}
+	if allCAs == nil {
+		return nil, fmt.Errorf("cannot use empty certificates pool")
+	}
+	extraCertFiles, err := filepath.Glob(filepath.Join(dirs.SnapdExtraSslCertsDir, "*.pem"))
+	if err != nil {
+		return nil, err
+	}
+	for _, p := range extraCertFiles {
+		extraCert, err := ioutil.ReadFile(p)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read extra certificate: %v", err)
+		}
+		// XXX: continue here trying to add others?
+		if ok := allCAs.AppendCertsFromPEM(extraCert); !ok {
```

Comment:
It's puzzling that the crypto package didn't chose to return an error here, just a boolean.